src/gradata/hooks/agent_precontext.py

```
@@ -0,0 +1,95 @@
+
+        brain_dir = resolve_brain_dir()
+        if not brain_dir:
+            return None
+
+        lessons_path = Path(brain_dir) / "lessons.md"
+        if not lessons_path.is_file():
+            return None
+
+        text = lessons_path.read_text(encoding="utf-8")
+        all_lessons = parse_lessons(text)
+        filtered = [l for l in all_lessons if l.state.name in ("RULE", "PATTERN") and l.confidence >= MIN_CONFIDENCE]
```

Ruff (0.15.9)

[error] Ambiguous variable name: `l`

(E741)